src/main/java/seedu/address/ui/TimetableWindow.java

```diff
@@ -1,41 +1,107 @@
 package seedu.address.ui;
 
+import java.net.URL;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.ResourceBundle;
 import java.util.logging.Logger;
 
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
 import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
 import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+import javafx.scene.text.Text;
 import javafx.stage.Stage;
 import seedu.address.commons.core.GuiSettings;
 import seedu.address.commons.core.LogsCenter;
 import seedu.address.logic.Logic;
 import seedu.address.logic.commands.CommandResult;
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.jobs.DeliveryJob;
+import seedu.address.ui.jobs.DeliveryJobListPanel;
 import seedu.address.ui.main.CommandBox;
 import seedu.address.ui.main.ResultDisplay;
 import seedu.address.ui.main.StatusBarFooter;
 
 /**
  * Controller for a timetable page
  */
-public class TimetableWindow extends UiPart<Stage> {
+public class TimetableWindow extends UiPart<Stage> implements Initializable {
 
     private static final String FXML = "TimetableWindow.fxml";
     private final Logger logger = LogsCenter.getLogger(getClass());
     private Stage primaryStage;
     private Logic logic;
+    private LocalDate focusDate;
+    private int focusDayOfWeek;
 
     // Independent Ui parts residing in this Ui container
     private ResultDisplay resultDisplay;
+
     @FXML
-    private StackPane commandBoxPlaceholder;
+    private VBox timetable;
+
+    @FXML
+    private Text year;
+
     @FXML
-    private StackPane jobListPanelPlaceholder;
+    private Text month;
+
+    @FXML
+    private StackPane commandBoxPlaceholder;
+
     @FXML
     private StackPane resultDisplayPlaceholder;
+
     @FXML
     private StackPane statusbarPlaceholder;
 
+    @FXML
+    private StackPane deliveryJobListPanelPlaceholder1;
+
+    @FXML
+    private StackPane deliveryJobListPanelPlaceholder2;
+
+    @FXML
+    private StackPane deliveryJobListPanelPlaceholder3;
+
+    @FXML
+    private StackPane deliveryJobListPanelPlaceholder4;
+
+    @FXML
+    private StackPane deliveryJobListPanelPlaceholder5;
+
+    @FXML
+    private StackPane deliveryJobListPanelPlaceholder6;
+
+    @FXML
+    private StackPane deliveryJobListPanelPlaceholder7;
+
+    @FXML
+    private Text day1;
+
+    @FXML
+    private Text day2;
+
+    @FXML
+    private Text day3;
+
+    @FXML
+    private Text day4;
+
+    @FXML
+    private Text day5;
+
+    @FXML
+    private Text day6;
+
+    @FXML
+    private Text day7;
```

Comment:
Maybe you can explore dynamic component assignment instead of static declaration, would be easier to maintain in the long run.

Reply:
hm ic will check that